### PR TITLE
Story/cite 183

### PR DESCRIPTION
**Pull request by PratikGiri**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
 

Link: https://diging.atlassian.net/browse/CITE-183
Description: For authors, editors, etc. there needs to be an import from the IsisCB

## Anything else the reviewer needs to know?

... describe here ...

### CODE REVIEW

File: citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    @Override
+    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize)
+            throws AuthorityServiceConnectionException {
+
+        String url = isiscbURL + isiscbSearchKeyword + this.getIsiscbSearchString(firstName, lastName)
+                + this.getLimitOffsetString(page, pageSize);
+
+        HttpHeaders isisCBheader = new HttpHeaders();
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java:108]: proper camelcase would be `isisCbHeader` or `isiscbHeader`

File: citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+                    authority.setName(isiscbEntry.getName());
+                    authority.setId(isiscbEntry.getId());
+                    authority.setUri(isiscbUriString.replace("{0}", isiscbEntry.getId()));
+                    authority.setDescription(isiscbEntry.getDescription());
+                    authorityEntries.add(authority);
+                }
+                searchResult.setFoundAuthorities(authorityEntries);
+                searchResult.setTotalPages((int) Math.ceil(isiscbEntries.getCount() / new Float(pageSize)));
+            }
+        } else {
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java:137]: if you turn the if-condition around, you can avoid the long if block.
```
if (response.getStatusCode() != HttpStatus.OK) {
    throw new ...
}
```

File: citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            isisCBSearchString = firstName == null ? lastName : firstName;
+        }
+        return isisCBSearchString;
+    }
+
+    private String getLimitOffsetString(int page, int pageSize) {
+        String isisCblimitOffsetString;
+        isisCblimitOffsetString = "&limit=" + pageSize + "&offset=" + pageSize * page;
+        return isisCblimitOffsetString;
```
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java:156]: merge all three lines into one

File: citesphere/src/main/resources/config.properties [hunk 4]
```diff
@@ -40,12 +40,16 @@ _viaf_authority_search_keyword=/viaf/search?query=
...
 
 _conceptpower_authority_search_keyword=ConceptSearch?word=
+_isiscb_authority_search_keyword=authority/search?query=
 
 _upload_files_directory=${upload.folder.path}
 
 _tasks_page_size=20
 _max_export_size=300
 _conceptpower_url=https://chps.asu.edu/conceptpower/rest/
+_isiscb_url=http://isiscb-staging-p3-2.us-west-2.elasticbeanstalk.com/rest/
```
Comment (jdamerow) [inline citesphere/src/main/resources/config.properties:50]: this needs to be configurable

File: citesphere/src/main/resources/config.properties [hunk 5]
```diff
@@ -40,12 +40,16 @@ _viaf_authority_search_keyword=/viaf/search?query=
...
+
+_isiscb_uri_string=https://data.isiscb.org/isis/authority/{0}/
```
Comment (jdamerow) [inline citesphere/src/main/resources/config.properties:52]: this needs to be configurable

File: citesphere/src/main/webapp/WEB-INF/views/auth/authorities/create.html [hunk 6]
```diff
@@ -231,6 +233,69 @@
     });
 }
 
+function getIsisCBAuthorities(modalType, personType, page) {
+    var firstName = $("#firstName" + personType).val();
+    var lastName = $("#lastName" + personType).val();
+    personType_lowerCase = personType.toLowerCase();
+    var authorizationToken = "Token 5e86a6fa0f3e18ea265cdc328c13018a6fc85cff";
```
Comment (jdamerow) [inline citesphere/src/main/webapp/WEB-INF/views/auth/authorities/create.html:240]: why is that here?

**Review by jdamerow — CHANGES_REQUESTED** [review]
Tokens should not be hard-coded and never ever pushed to a public repo!

File: citesphere/src/main/resources/config.properties (hunk 5 above)
Comment (PratikGiri) [reply citesphere/src/main/resources/config.properties:52]: So, I should move them in properties in pom.xml?

File: citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+    @Override
+    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize)
+            throws AuthorityServiceConnectionException {
+
+        String url = isiscbURL + isiscbSearchKeyword + this.getIsiscbSearchString(firstName, lastName)
+                + this.getLimitOffsetString(page, pageSize);
+
+        HttpHeaders isisCBheader = new HttpHeaders();
+        String isisCBtoken = "Token 5e86a6fa0f3e18ea265cdc328c13018a6fc85cff";
```
[the remark this answers is not in the document]
Comment (PratikGiri) [reply citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/IsiscbImporter.java:109]: Should I store it in DB and retrieve from there?
